proto_converter.go

```diff
@@ -114,12 +115,7 @@ func (converter *protoConverter) toField(key string, value interface{}) *cpb.Key
 		field.Value = &cpb.KeyValue_BoolValue{BoolValue: reflectedValue.Bool()}
 	default:
 		field.Value = &cpb.KeyValue_StringValue{StringValue: fmt.Sprint(reflectedValue)}
-		maybeLogInfof(
-			"value: %v, %T, is an unsupported type, and has been converted to string",
-			converter.verbose,
-			reflectedValue,
-			reflectedValue,
-		)
+		onEvent(newEventUnsupportedValue(key, value))
```

Comment:
This offends me. 😃 

Could we change it? (I know this is the old behavior.)
If the call to Sprint panics, then we might call it unsupported. Otherwise, it may implement fmt.Stringer in which case you're putting a spurious warning onto the user. If it prints but is not a Stringer, I will still argue it's supported based on the philosophy that it should not be an error to instrument things.

Reply:
I forgot to mention this! Rather than always throw this, I was wondering if we should explicitly typecast to `Stringer`, and record that value without a fuss. I would assume that recording the `Stringer` value was the intention of the caller. Likewise, we should check for the `error` interface, as that has a common and intentional string representation. We should only emit this event when we get something without any known common representation - and then record the `fmt.Sprint` value for it. 

Bear in mind, we are not "throwing an exception" or otherwise exploding in the users face, we are alerting the human in a safe way that they are doing something they probably did not intend to do. They can ignore these warnings if they don't care. If your argument is that it simply isn't an error to record an object, even an "unrecordable" one, as that was intentional on the part of the user – they want to introspect the damn thing, for example – then yes, I agree the warning should go away.  But bear in mind, the OT docs say: 

```
    // Tag values can be numeric types, strings, or bools. The behavior of
    // other tag value types is undefined at the OpenTracing level. If a
    // tracing system does not know how to handle a particular value type, it
    // may ignore the tag, but shall not panic.
```

So this event is to inform users that they are "out of compliance" with OT. My preference is that users should choose how they want to introspect an object when they record it:

```go
// This is an intentional representation; output is stable regardless of tracing vendor.
tracer.Tag("someObjTag", fmt.Sprintf("%T", someObj))

// In OT, you have no clue what data this will output – or even if the tag will be added.
tracer.Tag("someObjTag", someObj)
```

If we don't want to have this bikeshed, I can just eliminate this event for now, and add it in a subsequent PR.

Reply:
You make a good case. I'd like to not print this for fmt.Stringer, assuming that's intentional, otherwise fine and I retract my offended remarks.